controller-server/src/main/java/com/yahoo/vespa/hosted/controller/persistence/CuratorDb.java

```diff
@@ -739,6 +744,25 @@ public List<DataplaneTokenVersions> readDataplaneTokens(TenantName tenantName) {
         return readSlime(dataplaneTokenPath(tenantName)).map(DataplaneTokenSerializer::fromSlime).orElse(List.of());
     }
 
+    // -------------- Endpoint certificate pool -------------------------------
+
+    public void addToCertificatePool(String s, EndpointCertificateMetadata metadata, String pool) {
+        curator.set(certificatePoolRoot.append(pool).append(s), asJson(EndpointCertificateMetadataSerializer.toSlime(metadata)));
```

Comment:
If we model it like this we need to use use transactions to ensure atomicity when certificate state changes. However, if we instead store a `PooledCertificate` as a single object (see below), it can only ever have one state, so I suggest we do that.

Example:
`/controller/v1/certificatePool/<generated-identifier>`, which contains:
```
{
  "state": <pooled-certificate-state>,
  "certificate": <endpoint-certificate-metadata>,
}
```